src/sentry/static/sentry/app/components/forms/selectControl.tsx

```diff
@@ -32,12 +34,19 @@ const MultiValueRemove = props => (
 /**
  * Applies one set of styles onto the other while maintaining the same function
  * interface that is used by react-styled.
- * @param {*} newStyles the styles to apply on top of base
- * @param {*} baseStyles the style to override
+ *
+ * @param newStyles the styles to apply on top of base
+ * @param baseStyles the style to override
  */
-const combineStyles = (newStyles, baseStyles) => {
-  return Object.keys(newStyles || {}).reduce((computedStyles, key) => {
-    const styleFunc = (provided, state) =>
+const combineStyles = (
+  newStyles: StylesConfig | undefined,
+  baseStyles: StylesConfig
```

Comment:
```suggestion
  baseStyles: StylesConfig
  newStyles?: StylesConfig,
```

Reply:
I'm going to remove this function entirely. I found a similar function in react-select that we can import.